WordPress/Classes/ViewRelated/Gutenberg/GutenbergNetworking.swift

```diff
@@ -0,0 +1,62 @@
+import Alamofire
+
+struct GutenbergNetworkRequest {
+    typealias Response = (Swift.Result<Any, NSError>) -> Void
+
+    private let path: String
+    private unowned let blog: Blog
+
+    init(path: String, blog: Blog) {
+        self.path = path
+        self.blog = blog
+    }
+
+    func request(response: @escaping Response) {
+        if let dotComID = blog.dotComID {
+            dotComRequest(with: dotComID, response: response)
+        } else {
+            selfHostedRequest(response: response)
+        }
+    }
+
+    // MARK: - dotCom
+
+    private func dotComRequest(with dotComID: NSNumber, response: @escaping Response) {
+        blog.wordPressComRestApi()?.GET(dotComPath(with: dotComID), parameters: nil, success: { (responseObject, httpResponse) in
+            response(.success(responseObject))
+        }, failure: { (error, HTTPResponse) in
+            response(.failure(error))
+        })
+    }
+
+    private func dotComPath(with dotComID: NSNumber) -> String {
+        return path.replacingOccurrences(of: "/wp/v2/", with: "/wp/v2/sites/\(dotComID)/")
+    }
+
+    // MARK: - Self-Hosed
+
+    private func selfHostedRequest(response: @escaping Response) {
+        do {
+            let url = try blog.url(withPath: selfHostedPath).asURL()
+            performSelfHostedRequest(with: url, response: response)
+        } catch {
+            response(.failure(error as NSError))
+        }
+    }
+
+    private func performSelfHostedRequest(with url: URL, response: @escaping Response) {
+        SessionManager.default.request(url).validate().responseJSON { (responseObject) in
+            switch responseObject.result {
+            case .success(let responseObject):
+                response(.success(responseObject))
+            case .failure(let error):
+                response(.failure(error as NSError))
+            }
+        }
+    }
+
+    private var selfHostedPath: String {
+        let removedEditContext = path.replacingOccurrences(of: "context=edit", with: "context=view")
```

Comment:
Switching the context to `view` happens to work fine for this specific use case, but it likely won't for others. I don't like the idea of implementing it generically and pretending that `apiFetch` can now work for anything else, as it might lead to subtle undetected bugs.

Since I understand this is a temporary solution to an immediate problem, I think we should enforce that this system only works with a known set of API paths. So for any request other than getting media sizes, I would probably return a `Not implemented` error, or ignore the request and just log a warning.

Reply:
I think adding this kind of guard makes sense. Assuming we always expect that platforms to be in sync, it would be nice to have that guard on the JS-side so that we don't have to replicate it on both platforms.

Reply:
I thought on having some kind of `white list` of paths that we accept to avoid surprises.
This implementation also only do `GET` requests so far, so there's another reason for this. 👍
I agree that is probably better to do is on JS side, then probably this context change should also go on the JS side?

Reply:
>I agree that is probably better to do this on JS side, then probably this context change should also go on the JS side?

👍  The more path manipulation we move to the JS side, the better imo. I was planning to start investigating that once I got the Android implementation wrapped up.

Reply:
The SiteID is a WPCom thing, probably we don't want it on gutenberg repo.

Reply:
Good point @etoledom